packages/ui/src/components/va-form/VaForm.stories.ts

```diff
@@ -1,12 +1,286 @@
-import { defineComponent } from 'vue'
-import VaForm from './VaForm.demo.vue'
+import { userEvent } from '../../../.storybook/interaction-utils/userEvent'
+import { addText } from '../../../.storybook/interaction-utils/addText'
+import { expect } from '@storybook/jest'
+import { VaForm } from './'
+import { VaCheckbox } from '../va-checkbox'
+import { VaButton } from '../va-button'
+import { VaInput } from '../va-input'
+import { within } from '@storybook/testing-library'
 
 export default {
   title: 'VaForm',
   component: VaForm,
+  tags: ['autodocs'],
 }
 
-export const Default = defineComponent({
-  components: { VaForm },
-  template: '<VaForm/>',
+export const Default = () => ({
+  components: { VaForm, VaInput, VaButton },
+  template: `
+    <va-form ref="form">
+      <va-input :rules="[value => !!value || 'input is required']"/>
+    </va-form>
+    <va-button @click.prevent="$refs.form.validate()">
+      Submit
+    </va-button>
+  `,
 })
+
+export const Autofocus = () => ({
+  components: { VaForm, VaInput },
+  template: `
+    <va-form autofocus>
+      <va-input />
+    </va-form>
+  `,
+})
+
+addText(
+  Autofocus,
+  'Autofocus is broken',
+  'stale',
+)
+
+Autofocus.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
+  const input = canvas.getByRole('textbox', { name: '' }) as HTMLElement
+
+  await step('Focus first input', async () => {
+    expect(input).toHaveFocus()
+  })
+}
+
+export const Stateful = () => ({
+  components: { VaForm, VaCheckbox },
+  template: `
+    [true]
+    <va-form stateful>
+      <va-checkbox role/>
+    </va-form>
+    [false]
+    <va-form :stateful="false">
```

Comment:
The reason we want to keep stories like this looking like:
```
// for default false
va-form stateful
va-form
```
or
```
// for default true
va-form
va-form stateful=false
```

instead of 

```
va-form stateful=true
va-form stateful=false
```

is so that we can catch default change if someone does it in component. Default change could be considered a breaking change, so important to notice early.